### PR TITLE
update task-lib version

### PR DESCRIPTION
**Pull request by vineetmimrot**
Update task-lib version of following tasks:

1. AzureCLIV1
2. AzureResourceGroupDeploymentV2
3. AzureVmssDeploymentV0
4. DotNetCoreInstallerV1
5. GoToolV0
6. GoV0
7. PackerBuildV1
8. ServiceFabricComposeDeployV0
9. ServiceFabricDeployV1
10. ServiceFabricPowerShellV1
11. ServiceFabricUpdateManifestsV2

Rest of the task which Deployment-Azure team owns are already upgraded to latest task-lib version.

Note: PyPIPublisherV0 is not owned by us and also it is deprecated

### CODE REVIEW

File: Tasks/DotNetCoreInstallerV1/models.ts [hunk 1]
```diff
@@ -107,8 +106,7 @@ export class VersionParts {
                 throw "";
             }
 
-            semver.Range(version);
-            return true;
+            return !!semver.valid(version);
```
Comment (hiyadav) [inline Tasks/DotNetCoreInstallerV1/models.ts]: It should probably be => new semver.Range(version);

File: Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts [hunk 2]
```diff
@@ -42,9 +42,9 @@ let a: ma.TaskLibAnswers = <ma.TaskLibAnswers>{
 
 tr.setAnswers(a);
 
-tr.registerMock('vsts-task-lib/toolrunner', require('vsts-task-lib/mock-toolrunner'));
-tr.registerMock('azure-arm-rest/webRequestUtility', require('./mock_node_modules/webRequestUtility'));
-tr.registerMock('azure-arm-rest/azure-arm-compute', require('./mock_node_modules/azure-arm-compute'));
-tr.registerMock('azure-arm-rest/azure-arm-network', require('./mock_node_modules/azure-arm-network'));
-tr.registerMock('azure-arm-rest/azure-arm-resource', require('./mock_node_modules/azure-arm-resource'));
+tr.registerMock('azure-pipelines-task-lib/toolrunner', require('azure-pipelines-task-lib/mock-toolrunner'));
+tr.registerMock('azure-arm-rest-v2/webRequestUtility', require('./mock_node_modules/webRequestUtility'));
```
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts]: Why this change? azure-arm-rest-v2 might have breaking changes compated to azure-arm-rest

File: Tasks/AzureResourceGroupDeploymentV2/Tests/addVSTSExtension.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ tr.setInput("copyAzureVMTags", process.env["copyAzureVMTags"]);
 tr.setInput("deploymentGroupEndpoint", "PatEndpoint");
 tr.setInput("outputVariable", process.env["outputVariable"]);
 
+
```
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/Tests/addVSTSExtension.ts]: remove new line

File: Tasks/AzureResourceGroupDeploymentV2/make.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
     "common": [
         {
-            "module": "../Common/azure-arm-rest",
+            "module": "../Common/azure-arm-rest-v2",
```
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/make.json]: As mentioned earlier, I am am not sure this is right thing to do

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
+import path = require('path');
```
Comment (bishal-pdMSFT) [inline Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts]: Why create a new version of azure-blobstorage-artifactProvider? Does task lib introduce a breaking change in its behavior?

File: Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts (hunk 2 above)
Comment (vineetmimrot) [reply Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts]: azure-arm-rest-v2 contains upgraded version of azure-pipelines-task-lib. 
azure-arm-rest depends on vsts-tasks-lib & azure-arm-rest-v2 depends on azure-pipelines-task-lib.

So if you want to upgrade the task you should also use the upgraded common libraries which are using upgraded version of azure-pipelines-task-lib. o/w it may cause compatibility issues.

File: Tasks/AzureResourceGroupDeploymentV2/make.json (hunk 4 above)
Comment (vineetmimrot) [reply Tasks/AzureResourceGroupDeploymentV2/make.json]: same as above.

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts (hunk 5 above)
Comment (vineetmimrot) [reply Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts]: reasoning same as above.

File: Tasks/DotNetCoreInstallerV1/task.json [hunk 6]
```diff
@@ -13,8 +13,8 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 1,
-        "Minor": 0,
-        "Patch": 4
+        "Minor": 1,
```
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreInstallerV1/task.json]: Himanshu is also changing this. Check the version.

Comment (vineetmimrot) [reply Tasks/DotNetCoreInstallerV1/task.json]: Yeah, I have to check the version of all the modified tasks, as many of them have been upgraded since I have written this code.

File: Tasks/UseDotNetV2/task.json [hunk 7]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 2,
         "Minor": 0,
-        "Patch": 16
+        "Patch": 17
```
Comment (bishal-pdMSFT) [inline Tasks/UseDotNetV2/task.json]: Why update this task version if task lib is not updated

File: Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts (hunk 2 above)
Comment (bishal-pdMSFT) [reply Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts]: The goal here is to update task lib version. If you use v2 of common utilities they might have side impact of some common functionality being broken. And then we need to test for those.
Do you know what is delta between v1 and v2?

File: Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts (hunk 5 above)
Comment (bishal-pdMSFT) [reply Tasks/Common/azure-blobstorage-artifactProvider-v2/azureBlobStorageProvider.ts]: change in task lib does not necessarily mean new version of common libraries. Is the functionality being provided by the common lib change due to task lib version update?
By this logic, even task version should get major version update.

File: Tasks/UseDotNetV2/task.json (hunk 7 above)
Comment (vineetmimrot) [reply Tasks/UseDotNetV2/task.json]: dependent common/utility-common-2 has been changed

Comment (bishal-pdMSFT) [reply Tasks/UseDotNetV2/task.json]: Ohh yes

File: Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts (hunk 2 above)
Comment (vineetmimrot) [reply Tasks/AzureResourceGroupDeploymentV2/Tests/EnablePrereq.ts]: there is not much difference b/w azure-arm-rest and azure-arm-rest-v2. azure-arm-rest-v2 is using upgraded azure-pipelines-task-lib
